cmd/commandfactory.go

```diff
@@ -0,0 +1,28 @@
+package caddycmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type RootCommandFactory struct {
```

Comment:
If all this only gets used inside `caddycmd` we could just not export it? Is that fine? Or might other plugins/runtimes want to call this (thinking FrankenPHP, Caddy-Docker-Proxy)

Could use some godoc comments in here to explain the purpose.

Reply:
hmmmm. i don't think people should be importing it. if they need the snippet it they should copy the lines. i'll make it private.